### DISCUSSION TITLE
To XTouch Compact users

### DISCUSSION THREAD

**Discussion by MarcksHouse** [rsjaffe/MIDI2LR#745] [General]
Hie everybody,

Sorry for my English, I'm french I apology

I recently bouught an "Xtouch Compact" and I'm discovering the interface MIDI with Lightroom.

As I'm not aware of implementation, I found quit difficult to 'program' it, as I've to manage, discovaring the interface, its possibility and its limits, organising and adapting it to my workflow,

After some hard days, I dispose now, of one configuration that fits me in great lines.

After that job, I'd like to found other "X-Touch Compact" users, to share expérience, test configurations and improve existing one.
I'm sure there is several ways to organise so exchanges would be such a great benefit for me

Please to meet you and long life to the tool !

read you soon

**Comment by antzpantz** [comment]
👏👏👏 Welcome!

**Comment by omg-unicorns** [comment]
Hello, I also have the X-Touch Compact. I had been using PFixer 1.7.7 but that cosed-source project seems to have been abandoned, so I'm switching to this repo. I've only spent a few min verifying everything works. If anyone has already setup mappings that are similar to PFixer I'd be happy to have them and improve upon them. Thanks All!

**Reply by MarcksHouse** [reply]
Hi,

Here is the config I use at that time.

I had to construct it entirely by myself so, it's surely perfectible and adapted to my habbits and knowledge of LrC, but it may give you some ideas.

I'm affraid that a lots of explanations are in french but I'm sure you'll manage easely with a good traductor. If needed you can contact me for remarks or precisions on my choices.

Let me know what you think of it

Hoping that it may help you

Marc
[MDa V0 2023-10-16.zip](https://github.com/rsjaffe/MIDI2LR/files/12915369/MDa.V0.2023-10-16.zip)

**Comment by Papp-Anti** [comment]
Hello! I don't know English, so google translates! I just bought a compact too. I'm stuck on setting it up. The rsjaffe/midi2lr Wiki says so "Go to Files→Plugin Extras→MIDI2LR Keywords. MIDI2LR starts indexing your keywords as soon as Lightroom is started. If you have thousands of keywords, it may take a minute or two. If you open this menu before it is all indexed, you'll get a message telling you how many keywords have been indexed so far, and you'll then receive a bezel notification when the keyword indexing is completed. " On the other hand, nothing loads for me, no matter how long I wait. The keyword fields are empty. Actually, I don't know if I can set it here so that I can assign watermarks or user export profiles to buttons on the midi controller. does anyone have any ideas? Thank you for helping me!

**Reply by MarcksHouse** [reply]
Hie, Papp-Anti,

Do you speak french ?
I don't use Keywords very much and I don't know if MIDI2Lr is able to manage with all my export configrations,

When I open the MIDI2Lr menu/keywords the first ones are indexed:
![Capture d'écran 2024-01-15 182953](https://github.com/rsjaffe/MIDI2LR/assets/144238913/32c5b198-ee52-4a54-bf40-9e1ffff4fc6e),

I did nothing for this !

sorry If I can't do more for you

**Comment by xetyr** [comment]
Hey @MarcksHouse,

Thank you, you opened an interesting topic.

I'm also a new user of the X-Touch compact. I previously used the BCF-2000, but the noise of faders was a plague and I was not able to accept it anymore.

I don't regret my choice, this controller has almost everything I'm needing !
Noise isn't so dramatic, especially compared to my old controller haha.


Well, I also had a look at your config sample, because I still trying to find the ideal config, and your suggestion is interesting.
Like you, I intent to use the buttons 

The major interest of the BCF-2000 was the presence of buttons "STORE", "LEARN", "EDIT" and "EXIT".
It was allowing me to switch between T, S and L and control each color canal knobs. It was like an embedded profiles on the controler.
You found a trick with your different .XML files, and I will use them too. My only regret is to not be allowed to keep the corresponding light button on. I tried to change the behavior of them to "toggle" on the X-Touch editor. Light is fixed on but it's not working because (I guess) the controller is sending an active signal to the Lr.


So, I would like to thank you for this topic, it's a nice base for X-Touch users. And thank you too for your template, I will use it to understand how profiles works, because I never tried them.

P.S. Thank you also for your Word and Excel files, it's explicit. I probably found a mistake on your sheet "Console type", cell A25: A>B
One question about your tuto : why did you set shortcuts ? MIDI2LR is able to directly assign these commands isn't it ?

**Reply by MarcksHouse** [reply]
Thank you xetyr for those words.
I'm pleased that my post inspire you.

First, I haven't found either the way of unlighting buttons, and would appreciate it, but I think it's not possible, because LrC didn't generate the return signal (to be verified)

Then, I use shortcuts when I haven't found the appropriate command. Newer version may have improve this. I'll have to check.

So long
Marc

**Comment by xetyr** [comment]
Hi there,

I'm pleased to share with you my X-Touch config (V1.1) !

On my [repo](https://github.com/xetyr/MIDI2LR-X-Touch-Compact-config) you'll find :
• Layers A and B for X-Touch config through Behringer's software (identical to Marck's one) ;
• 1 Excel document recapitulating each config file ;
• 4 XML files, according to my setup ;
• 1 printable layout to stick on the controller, with labels. I printed it on a self-sticking paper, a must-have !

The Excel document is hugely inspired by @MarcksHouse, adapted to my needs. I'm not sharing his installation tutorial, because I guess MIDI2LR users know how to try it.

Short explanations :
• Based on 4 profiles : 1.Library  2. Development  3.HSL tools  4.ColorGrading ;
• Redundancies between 2.Dev and 3.HSL because I'm using them together, especially with Lum/Sat channels (see A and B layers) ;
• Mix of frenglish on Excel and on tiff template because I'm sorry frenchie ;
• Well commented XMLs files, in french.

I'll be curious to have feedback on it. I need more practice before a new version, maybe I'll release a V1.2 if you are interested in.

Have a nice day

**Reply by MarcksHouse** [reply]
Thanks you @xetyr,
happy that I'd helped you to begin,
Marc

**Reply by yodavidverde** [reply]
Hi folks,
 Im probably going to buy the Xtouch Compact as well and your files! I was wondering, can you use Xtouch to adjust the settings for a mask too (such as exposure, clarity, etc) or is it hard to switch from the general controls to mask controls? I assume you still need to use the mouse frequently for mask? What would you guys say are the biggest cons to using the Xtouch compared to mouse and keyboard? 
thanks,
Best,
David

**Reply by xetyr** [reply]
hello @yodavidverde,
 You're asking a good question !

**1. Masks abilities**

As far as I know, each mask command could be mapped through MIDI2LR (see [here](https://github.com/rsjaffe/MIDI2LR/wiki/Commands#23)), but there is nothing to set the start point and the direction.

For me, it means you can create masks (e.g. "Add sky to the mask") and navigate through them, but I suppose you still need to adjust mask values and "mouse things" manually. In the end, I bet you can use a hybrid workflow with controllers + mouse.
With my usage, it seems to be a waste of time because my hand is already on the mouse when creating/adjusting masks, this is why I did not even try to work like this. The controller is fun, but also a time saver when editing my professional photographs, it's counterproductive if I spend time switching from the controller to the mouse, while I can reach the same objectives with few clicks...


**2. Pros & Cons**

The biggest cons are about programming the controller. It's not very complicated, but you'll pass some time to understand preset configurations like MarksHouse and mine. You'll probably find these keybindings could be optimized, according to your habits in editing. So you'll spend time customizing or creating your own profiles. It's the most time-consuming phase. But as soon as you find your configuration, you'll be faster than ever when editing, and you be able to fine-tune your images keeping your eyes on the screen.
An important point : your eyes are an excellent comparator but a bad analyzer. You're keeping your eyes on your image while editing, because you'll know each knob with practice. It's a more efficient way to work, without looking at the right panel before each adjust.



To conclude, I'll say :
• using a controller is a time saver ;
• it allows you to work with an incredible flexibility, using every tool without scrolling and searching things ;
• biggest con is the learning curve, a controller is a new tool, anew way to work and you'll need to get familiar with it ;
• customization of the controller is a freedom, but you'll to find your ergonomics, adding stickers on it, for example.

By the way, welcome to the world of MIDI2LR, it's a pretty nice approach of photo editing. The Xtouch Compact is, as far as I know, one of the most interesting price-quality ratios.

**Reply by yodavidverde** [reply]
Xetyr, thanks so much for your thought-out and detailed answer!  I am excited to join the world of MIDI2LR for multiple reasons, but mainly to have a more ergonomic work flow that also saves time. Im trying to save my wrists from the repetitive strain of mouse usage and I think this is the way! Im debating between a Behringer and an Akai at the moment.
Best regards,
David
